x-pack/plugins/cloud/public/plugin.tsx

```diff
@@ -268,21 +268,31 @@ export class CloudPlugin implements Plugin<CloudSetup> {
               user.authentication_realm?.name === 'cloud-saml-kibana'
             ) {
               // If authenticated via Cloud SAML, use the SAML username as the user ID
-              return user.username;
+              return { userId: user.username, isElasticCloudUser: true };
```

Comment:
Can't a user be on cloud and authenticate using a different method (i.e. user name)?

Reply:
Not at the moment. Currently, Cloud users are only authenticated via `cloud-saml-kibana` (refer to https://github.com/elastic/kibana/issues/129236 for a description of when we can assume a user is an Elastic Cloud authenticated user). This may change in the future though as Cloud allows new authentication providers.

We may need to revisit the preceding `if` clause at that point. Although the security team may do the heavy-lifting for us (refer to changes in https://github.com/elastic/kibana/pull/132622/files#diff-d6ea621c5aa5bbb2c2aeeef7a65f511fecbd81282fc2f293c76db5f36752af12R265) as they are exposing a new flag in the response of `getCurrentUser`.

Reply:
Thanks for explaining
I was just thinking of having a better name for the parameter?

Reply:
Perhaps we could change `isElasticCloudUser` to `isElasticCloudSSO`? To @lizozom's point, someone authenticating via basic auth into Kibana is technically still a Cloud user (because they are... using Elastic Cloud) but their auth method is not Cloud-native.

Reply:
++ to renaming the field. However, I wonder if `isElasticCloudSSO` is future-proof. Elastic Cloud will support other authentication providers in the future and it's still unclear whether Kibana will see them as _Elastic Cloud **Authenticated**_ users or if the actual realms will be forwarded to Kibana. Perhaps `isAuthenticatedViaElasticCloud` is more descriptive?

If in the future, Kibana will be able to differentiate when a user is authenticated via Elastic Cloud but not Elastic Cloud's SSO, we may be able to claim `isAuthenticatedViaElasticCloud: true` and `isElasticCloudSSO: false`.

FWIW, when choosing `isElasticCloudUser`, I only followed the naming convention from https://github.com/elastic/kibana/pull/132622/files#diff-d6ea621c5aa5bbb2c2aeeef7a65f511fecbd81282fc2f293c76db5f36752af12R265. 

Probably @elastic/kibana-security can shed some light on what the plans are so we can find the best descriptive name?

Reply:
I checked on Slack with the @elastic/kibana-security team and their response could be summarized in this quote:

> The data I believe you're trying to capture is "Is this Kibana User actually a user managed by ESS?"
You don't care how they authenticated, you only care that they are a cloud-managed user.
The other proposed names try to complicate things by including irrelevant information.

@lizozom @pgayvallet @linyaru what do you think? Should we start off with `isElasticCloudUser` as in "Is it a user _owned_ by ESS?" and add more info re the authentication provider in the future if we see fit?